File: Python/572. Subtree of Another Tree.py
class Solution:
    
    def dfs(self, s, t):
        if s is None and t is None:
            return True
        elif s is None or t is None:
            return False
        if s.val != t.val:
            return False
        return self.dfs(s.left, t.left) and self.dfs(s.right, t.right)
        
    def isSubtree(self, s, t):
        """
        :type s: TreeNode
        :type t: TreeNode
        :rtype: bool
        """
        if s is None and t:
            return False
        if self.dfs(s, t):
            return True
        print(s.val)
        return self.isSubtree(s.left, t) or self.isSubtree(s.right, t)
        

File: C++/572. Subtree of Another Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    bool dfs(TreeNode* s, TreeNode* t)
    {
        if(s==nullptr || t==nullptr)
            return s == t;
        if(s->val != t->val)
            return false;
        return dfs(s->left, t->left) && dfs(s->right, t->right);
        
    }
    bool isSubtree(TreeNode* s, TreeNode* t) {
        if(s==nullptr)
            return s==t;
        if(dfs(s, t))
            return true;
        return isSubtree(s->left, t) || isSubtree(s->right, t);
    }
};